src/main/java/fi/aalto/cs/apluscourses/ui/courseproject/CourseSelectionView.java

```diff
@@ -82,6 +87,11 @@ public void focusGained(FocusEvent focusEvent) {
 
       @Override
       public void focusLost(FocusEvent focusEvent) {
+        try {
+          throw new OperationNotSupportedException();
+        } catch (OperationNotSupportedException e) {
+          logger.info("Operation not supported", e);
+        }
       }
```

Comment:
Do you throw and catch to get the stacktrace to the log?

Still, this makes very little sense to me.

Why we have to log events we don't care about? I guess Sonar complained of an empty method body, right? That can be solved by adding a comment there. This method should do nothing, IMO.